### PR TITLE
Jenkinsfile improvements

### PR DESCRIPTION
**Pull request by danberindei**
* Split compilation into a separate stage
* Extract MAVEN_HOME definition outside stages
* Add warnings step, parse Maven and Javac warnings
* Add checkstyle step
* Add healthScaleFactor: 100

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -20,27 +23,32 @@ pipeline {
         stage('Build') {
             steps {
                 configFileProvider([configFile(fileId: 'maven-settings-with-deploy-snapshot', variable: 'MAVEN_SETTINGS')]) {
-                    script {
-                        def mvnHome = tool 'Maven'
-                        sh "${mvnHome}/bin/mvn clean install -s $MAVEN_SETTINGS -Dmaven.test.failure.ignore=true || true"
-                        junit testDataPublishers: [[$class: 'ClaimTestDataPublisher']], testResults: '**/target/*-reports/*.xml'
-                        sh "${mvnHome}/bin/mvn clean -s $MAVEN_SETTINGS"
+                    timeout (time = 2, unit = 'HOURS') {
```
Comment (slaskawi) [inline Jenkinsfile]: I proposed global timeout here: https://github.com/infinispan/infinispan/pull/5109/files

WDYT?

Comment (danberindei) [inline Jenkinsfile]: Sure. I mainly wanted to experiment with block timeout in order to put a tighter limit on the stress tests, but I can't find a way to pass a timeout unit :)

**Comment by gustavonalle** [comment]
@danberindei needs rebase, because I was not aware you had fixed the jcache TCK reporting here before sending https://github.com/infinispan/infinispan/pull/5177, sorry for that

**Comment by danberindei** [comment]
Np @gustavonalle, the PR already had some problems because I haven't been keeping it up to date.

**Comment by galderz** [comment]
Thx @danberindei

**Pull request merged**